Adding documentation for default task resolve checking method (#466)
medic/cht-docs: content/en/apps/reference/tasks.md

};
```

### Default resolvedIf method

> If you skip defining `resolvedIf` section in the task definition, it will default to the `defaultResolvedIf` method. You can only skip the section if you have at least one action with `report` type.

Reviewer: ```suggestion
If the `resolvedIf` is undefined in an action of type `report`, then `resolvedIf` is going to default to `defaultResolvedIf` function.
```

> The `defaultResolvedIf` method looks for reports assigned to the contact, and returns `true` if it finds any report that matches the `form` defined in your first action with `report` type. Only the reports submitted during a specific time period are considered:

Reviewer: ```suggestion
The `defaultResolvedIf` method returns `true` if it finds any report assigned to the contact that matches the `form` defined in the action of type `report`. Only the reports submitted during a specific time period are considered:
```

};
```

## Default resolvedIf method

> If you skip defining `resolvedIf` section in the task definition, it will default to defaultResolvedIf method which is equivalent to this:

Reviewer: Maybe explaining the why, that if actions type is `report` and `resolvedIf`isn't defined then `defaultResolvedIf` will be executed in its place. 

Also that `defaultResolvedIf` is available in the task scope to be used and how to use it (code sample).

You can also explain with few words what `defaultResolvedIf` validates (conditions|rules) so it's clear the outcome, no need of pasting it's code since it can change slightly and be out of date easy, better that if someone is interested of looking at the implementation then they can go see the open source. 
Specially because `defaultResolvedIf` won't be modified from outside but just used.

```js
resolvedIf: function (c, r, event, dueDate) {
  let start = 0;

  //The first report type action's form 
  var resolvingForm = taskDefinition.actions.find(action => action.type === 'report').form;

  if (r) {//Report based task
    //Either "start of the task window" or "after the report's reported date", whichever comes later
    start = Math.max(Utils.addDate(dueDate, -event.start).getTime(), r.reported_date + 1);
  }
  else {//Contact based task
    //Start of the task window
    start = Utils.addDate(dueDate, -event.start).getTime();
  }

  //End of the task window
  const end = Utils.addDate(dueDate, event.end + 1).getTime();

  //Returns true if the contact has at least one report with matching form type
  //that is submitted between start and end times (inclusive)
  return Utils.isFormSubmittedInWindow(
    c.reports,
    resolvingForm,
    start,
    end
  );
},
```

You can also use `this.defaultResolvedIf` directly:

```js
resolvedIf: this.defaultResolvedIf,//Same as skipping resolvedIf
```

It is also possible to specify a resolving form type:

```js

> resolvedIf: function (c, r, event, dueDate) {

Reviewer: Please replace 'c' for 'contact' and 'r' for 'report', trying to fix the one letter names for more descriptive names. Thanks!

Remember to add semi-colon at the end of each JS line. 

And use `cons` and `let` instead of `var`, basically leaving the best best practices in the code sample for others to replicate 💪
